### PR TITLE
Prefetching dependencies

### PR DESCRIPTION
**Pull request by missmesss**
The text is being reviewed and will probably see a bit of changes.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 1]
```diff
@@ -1,7 +1,251 @@
 = Prefetching package manager dependencies for hermetic builds
 
+//NOTE: We may want to rephrase this to word for more accurate SBOM information
+//NOTE: Should this be a subtopic for hermetic or a sibling to it?
+
+In {ProductName}, you can run a hermetic build by restricting network access to the build, but without network a build can’t fetch component dependencies from a repository and might fail. To avoid that, {ProductName} can prefetch dependencies for your hermetic builds using link:https://github.com/containerbuildsystem/cachi2/blob/main/README.md[Cachi2].
+
+For every build, Cachi2 generates a software bill of materials (SBOM) where all dependencies are properly declared and pinned to specific versions. Also, Cachi2 ensures that arbitrary code is never executed during the prefetch, meaning, for example, that the build doesn’t pull any undeclared dependencies. Such measures result in very accurate SBOMs and improve the build reproducibility. For more information about SBOMs, see link:https://konflux-ci.dev/docs/how-tos/metadata/sboms/[Inspecting SBOMs].
```
Comment (missmesss) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:8]: @arewm @brunoapimentel Will you please review the introduction on lines 6 and 8? Following [Bruno's comment in a Gdoc](https://docs.google.com/document/d/18jPr7v1w-uOnsh6hM2tUaH47ctpWZgjsxeCJGHY_DLs), I've reworded this intro to add a bit more info about benefits that prefetching and Cachi2 give. WDYT?

The previous version said:

> In Konflux, you can run a hermetic build by restricting network access to the build. Consequently, a build might fail if it can’t retrieve dependencies from a repository. To address this, Konflux uses Cachi2 to prefetch package manager dependencies for supported languages. For every build, Cachi2 also generates a software bill of materials (SBOM) of all dependencies included in your builds, for better transparency and build maintainability. For more information about SBOMs, see [Inspecting SBOMs](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/Secure-your-supply-chain/proc_inspect_sbom/).

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 2]
```diff
@@ -1,7 +1,251 @@
 = Prefetching package manager dependencies for hermetic builds
 
+//NOTE: We may want to rephrase this to word for more accurate SBOM information
+//NOTE: Should this be a subtopic for hermetic or a sibling to it?
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:4]: I think a sibling. There may be use cases where prefetching is beneficial even without hermetic builds.

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc (hunk 1 above)
Comment (arewm) [reply docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:8]: seems reasonable to me

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 3]
```diff
@@ -1,7 +1,251 @@
...
+== Enabling prefetch builds for `gomod`
+
+.Prerequisites
+* You have a `go.mod` file in your repository that lists all the dependencies.
+
+.Procedure
+To create a hermetic build for a component, complete the following steps:
+
+. Go to the `.tekton` directory in your component repository and find the `.yaml` files related to both the `*pull request*` and `*push*` processes.
+. To configure the hermetic pipeline in both `.yaml` files, add the following hermetic pipeline parameters to the `spec.params` sections:
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:38]: We are not enabling hermetic here.

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 4]
```diff
@@ -1,7 +1,251 @@
...
+----
+<1> The `*prefetch-input*` parameter specifies the path to the directory that has the lockfile and the package metadata files. In this example, the `.` indicates that the package manager lockfile is in the repository root. Additionally, if you have multiple directories, you can provide the path to those directories in the JSON array format. For example, `[{"path": ".", "type": "gomod"}, {"path": "subpath/to/the/other/directory", "type": "gomod"}]`.
+
+. Create a pull request by committing your changes to the repository of the component.
+
+. Review and merge the pull request.
+
+.Verification
+* From the {ProductName} *Applications* view, go to *Activity > Pipeline runs*.
+** Go to the pipeline run with *Build* in the *Type* column and confirm that the `pre-fetch dependencies` stage displays a green checkmark. This indicates that the build process successfully fetched all dependencies.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:57]: The task name is `prefetch-dependencies`

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 5]
```diff
@@ -1,7 +1,251 @@
...
+
++
+[NOTE]
+====
+* By default, Cachi2 processes `requirements.txt` and `requirements-build.txt` at a specified path.
+
+* When adding these parameters, you can safely ignore the default values for the link:https://github.com/burrsutter/partner-catalog-stage/blob/e2ebb05ba8b4e842010710898d555ed3ba687329/.tekton/partner-catalog-stage-wgxd-pull-request.yaml#L90[`pipelineSpec.params`] in the `.yaml` files.
+====
+
+.. Optional: For requirements files without default names and path, add the following hermetic pipeline parameters in both the `.yaml` files:
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:185]: These parameters are prefetch-input.

**Review by arewm — CHANGES_REQUESTED** [review]
The hermetic and prefetch-dependencies comments are relevant in multiple places but I didn't replicate the comment.

Cachi2 also supports yarn and prefetching pip wheels. How complete do we want this documentation vs. just referring to https://github.com/containerbuildsystem/cachi2/blob/main/docs/usage.md ?

**Comment by missmesss** [comment]
@arewm I've added the changes requested by you but I understand you won't see them before the vacation.

I plan to merge this PR because the changes are quite straightforward (like updating parameters names).

**Comment by missmesss** [comment]
> Cachi2 also supports yarn and prefetching pip wheels. How complete do we want this documentation vs. just referring to https://github.com/containerbuildsystem/cachi2/blob/main/docs/usage.md ?

Thank you for pointing this out.
Usually we aim at not repeating (copypasting) docs that already exist. However, in case of this guide it makes sense to extend it with a section about Yarn. I'll open an issue about it and discuss it with my team.

**Comment by missmesss** [comment]
Applied all requested changes. Since they're quite straightforward, e.g. changing parameters names, re-review isn't strictly needed, I'm merging this PR.
